procurement_portal/records/views.py

```diff
@@ -78,14 +97,42 @@ def _get_xlsx_url(self, request):
         )
 
 
-class PurchaseRecordXLSXListView(XLSXFileMixin, BasePurchaseRecordListView):
+class PurchaseRecordXLSXTemplateListView(XLSXFileMixin, BasePurchaseRecordListView):
```

Comment:
This view does ensure types are set, doesn't it? Or is it actually just about dynamically generating column names?

If it's about column names, I think there's a simpler way, like iterating over purchase record field names.

```suggestion
class PurchaseRecordXLSXTemplateListView(XLSXFileMixin, BasePurchaseRecordListView):
    """Use DRF XLSX view to dynamically generate an excel file with the correct types in each column"""
```